src/main/java/ch/njol/skript/events/EvtDamage.java

```diff
@@ -39,34 +39,51 @@
 @SuppressWarnings("unchecked")
 public class EvtDamage extends SkriptEvent {
 	static {
-		Skript.registerEvent("Damage", EvtDamage.class, EntityDamageEvent.class, "damag(e|ing) [of %entitydata%]")
+		Skript.registerEvent("Damage", EvtDamage.class, EntityDamageEvent.class, "damag(e|ing) [of %entitydata%] [by %entitydata%]")
 				.description("Called when an entity receives damage, e.g. by an attack from another entity, lava, fire, drowning, fall, suffocation, etc.")
-				.examples("on damage:", "on damage of a player:")
-				.since("1.0");
+				.examples("on damage:", "on damage of a player:", "on damage of player by zombie:")
+				.since("1.0, INSERT VERSION for by entitydata");
 	}
 	
 	@Nullable
-	private Literal<EntityData<?>> types;
+	private Literal<EntityData<?>> ofTypes, byTypes;
 	
 	@Override
 	public boolean init(final Literal<?>[] args, final int matchedPattern, final ParseResult parser) {
-		types = (Literal<EntityData<?>>) args[0];
+		ofTypes = (Literal<EntityData<?>>) args[0];
+		byTypes = (Literal<EntityData<?>>) args[1];
 		return true;
 	}
 	
 	@Override
 	public boolean check(final Event evt) {
 		final EntityDamageEvent e = (EntityDamageEvent) evt;
-		if (!checkType(e.getEntity()))
+		if (evt instanceof EntityDamageByEntityEvent) {
+			EntityDamageByEntityEvent event = (EntityDamageByEntityEvent) evt;
+			if (!checkDamager(event.getDamager()))
+				return false;
+		}
+		if (!checkDamaged(e.getEntity()))
 			return false;
 		if (e instanceof EntityDamageByEntityEvent && ((EntityDamageByEntityEvent) e).getDamager() instanceof EnderDragon && ((EntityDamageByEntityEvent) e).getEntity() instanceof EnderDragon)
 			return false;
 		return checkDamage(e);
 	}
+
+	private boolean checkDamager(final Entity e) {
```

Comment:
```suggestion
	private boolean checkDamager(Entity e) {
```
Same goes for all other parameters and variables

Reply:
just tried to keep with the standard of the class, ill change them